openedx/core/djangoapps/catalog/management/commands/cache_programs.py

```diff
@@ -188,3 +203,28 @@ def process_pathways(self, site, pathways, programs):
                 logger.exception('Failed to process pathways for {domain}'.format(domain=site.domain))
                 failure = True
         return processed_pathways, programs, failure
+
+    def get_courses(self, client, site):
+        """
+        Get all courses for the current client
+        """
+        failure = False
+        courses = {}
+        try:
+            logger.info('Requesting courses for {domain}.'.format(domain=site.domain))
+            next_page = 1
+            while next_page:
+                # Emma: hidden? as a parameter
+                new_courses = client.course_runs.get(exclude_utm=1, page=next_page) 
+                courses.extend({cr[course]:cr[programs] for cr in new_courses['results']})
```

Comment:
1. dictionary keys need to be strings, not variables, in this case.
2. I think you just want the program uuids, right, not the full program details? Is that what `cr['programs']` is?

Reply:
1. It is a variable that represents a string. I was pretty sure that was a thing I could do... Maybe we need to talk in person about it.
2. Do I want the program uuids? If I have the information on it anyhow?